StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift

```diff
@@ -238,6 +227,11 @@ class EmbeddedPaymentMethodsView: UIView {
             """)
             return
         }
+        
+        // Remove old mapping from selectionButtonMapping
+        if let previousSelection = selectionButtonMapping.first(where: { $0.value == previousSavedPaymentMethodButton })?.key {
+            selectionButtonMapping.removeValue(forKey: previousSelection)
+        }
```

Comment:
This function is only called when there was a previouslySavedPaymentMethod button, which, if i understand correctly means that there must exist a mapping in `selectionButtonMapping`, correct?

If so, seems like `let previousSelection = selectionButtonMapping.first(where..` should be part of the guard statement?

Reply:
Yep I think you're right, I don't think this would ever be called when we don't have an existing mapping to a saved button. This step is pretty much optional/nice-to-have/clean up that we should do. I think we could actually get away w/o doing it at all.

Since it's mostly optional, I don't feel like it should block the execution of the rest of the function, even though I can't think of a state where that would happen. Feels slightly safer outside the guard to me.

Reply:
If this function's precondition is to have `self.savedPaymentMethodButton` set, it seems that there would be a bug if there was no entry in selectionButtonMapping, correct?

Seems like moving it to the guard would strengthen the guard condition and has the potential to catch bugs if this condition is no longer being met.

Reply:
Hrm, I don't feel strongly. I'll add it.